### PR TITLE
beta-stage-----> production (beta merge with production & resolved conflicts)

### PR DESCRIPTION
**Pull request by rohit-walkover**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.ts [hunk 1]
```diff
@@ -380,11 +381,16 @@ export class NewLedgerEntryPanelComponent implements OnInit, OnDestroy, OnChange
...
             activeAccountTaxes = this.activeAccount.applicableTaxes.map((tax) => tax.uniqueName);
+            if(this.activeAccount.otherApplicableTaxes && this.activeAccount.otherApplicableTaxes.length && activeAccountTaxes && activeAccountTaxes.length) {
+                if(this.activeAccount.otherApplicableTaxes[0].uniqueName !== activeAccountTaxes[0] && activeAccountTaxes.includes(this.activeAccount.otherApplicableTaxes[0].uniqueName)) {
+                  activeAccountTaxes = activeAccountTaxes.reverse();
+                }
+            }
         }
         if (this.currentTxn.selectedAccount.stock && this.currentTxn.selectedAccount.stock.stockTaxes && this.currentTxn.selectedAccount.stock.stockTaxes.length) {
-            this.taxListForStock = this.mergeInvolvedAccountsTaxes(this.currentTxn.selectedAccount.stock.stockTaxes, activeAccountTaxes);
-        } else if (this.currentTxn.selectedAccount && this.currentTxn.selectedAccount.parentGroups && this.currentTxn.selectedAccount.parentGroups.length) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.ts:386]: don't revert this change

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.ts [hunk 2]
```diff
@@ -1487,14 +1497,27 @@ export class NewLedgerEntryPanelComponent implements OnInit, OnDestroy, OnChange
         if (this.currentTxn && this.currentTxn.selectedAccount && this.currentTxn.selectedAccount.accountApplicableDiscounts && this.currentTxn.selectedAccount.accountApplicableDiscounts.length) {
             this.currentTxn.selectedAccount.accountApplicableDiscounts.map(item => item.isActive = true);
             this.currentTxn.discounts.map(item => { item.isActive = false });
-            this.currentTxn.selectedAccount.accountApplicableDiscounts.forEach(element => {
-                this.currentTxn.discounts.map(item => {
-                    if (element.uniqueName === item.discountUniqueName) {
-                        item.isActive = true;
-                    }
-                    return item;
+            if(this.currentTxn.discounts && this.currentTxn.discounts.length === 1) {
```
Comment (prateek-giddh) [inline apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.ts:1500]: We are performing same logic in if else please check

Comment (rohit-walkover) [reply apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.ts:1500]: Yes but both will execute for diff condition
if there is newLedger entry panel discount not initi. then there will be 1 discount by default for value/percent in that case we need to update the discount array after discount componenet get init we will execute discount update array at last so implemented it with setTime so it get execute at last irrespective to component lifecycle

**Review by ravinderwalkover — APPROVED** [review]

**Review by prateek-giddh — APPROVED** [review]
